[llava][8/N] Add Multimodal Runner

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #4426
* __->__ #4354

Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

larryliu0820 committed:
Add Multimodal Runner

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"




[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"




[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D60101911).

kirklandsign commented on examples/models/llava/runner/CMakeLists.txt:
```diff
@@ -0,0 +1,76 @@
+       ${CMAKE_CURRENT_SOURCE_DIR}/../../../../extension/llm/tokenizer/tiktoken.cpp
+  )
+  list(APPEND _multimodal_runner__srcs
+       ${CMAKE_CURRENT_SOURCE_DIR}/../tokenizer/llama_tiktoken.cpp
```
${CMAKE_CURRENT_SOURCE_DIR}/../../llama2/tokenizer/llama_tiktoken.cpp
?

larryliu0820 replied:
Yeah I'll probably delete this

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D60101911).

iseeyuan commented on backends/xnnpack/serialization/runtime_schema.fbs:
```diff
@@ -134,6 +134,7 @@ union XNodeUnion {
   XNNConcatenate4: _XNNCat,
   XNNStaticSlice,
   XNNScaledDotProductAttention,
+  XNNBatchMatrixMultiply: _XNNNode2x1,
```
Are those be landed in a separate PR?

larryliu0820 replied:
Yeah

larryliu0820 committed:
Update on "[llava][8/N] Add Multimodal Runner"


Differential Revision: [D60101911](https://our.internmc.facebook.com/intern/diff/D60101911)

[ghstack-poisoned]

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D60101911).
